service-base/src/main/java/org/eclipse/hono/service/EventBusService.java

```diff
@@ -176,17 +176,23 @@ private void processRequestMessage(final Message<JsonObject> msg) {
      * @return The property value or {@code null} if no such property exists or is not of the expected type.
      * @throws NullPointerException if any of the parameters is {@code null}.
      */
-    @SuppressWarnings({ "unchecked" })
-    protected final <T> T getTypesafeValueForField(final JsonObject payload, final String field) {
+    protected static final <T> T getTypesafeValueForField(final Class<T> clazz, final JsonObject payload,
+            final String field) {
 
+        Objects.requireNonNull(clazz);
         Objects.requireNonNull(payload);
         Objects.requireNonNull(field);
 
-        try {
-            return (T) payload.getValue(field);
-        } catch (final ClassCastException e) {
+        final Object result = payload.getValue(field);
+        if (result == null) {
```

Comment:
This check is unnecessary, the `clazz.isInstance(result)` will return false if the object is null.

Reply:
It indeed is.